src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php

```diff
@@ -0,0 +1,290 @@
+ * @author Egidijus Girčys <e.gircys@gmail.com>
+ */
+final class MultilineWhitespaceBeforeSemicolonsFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * @internal
+     */
+    const STRATEGY_NO_MULTI_LINE = 'no_multi_line';
+
+    /**
+     * @internal
+     */
+    const STRATEGY_NEW_LINE_FOR_CHAINED_CALLS = 'new_line_for_chained_calls';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(';');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Forbid multi-line whitespace before the closing semicolon or move the semicolon to the new line for chained calls.',
+            [
+                new CodeSample(
+                    '<?php
+function foo () {
+    return 1 + 2
+        ;
+}
+'
+                ),
+                new CodeSample(
+                    '<?php
+                        $this->method1()
+                            ->method2()
+                            ->method(3);
+                    ?>
+',
+                    ['strategy' => self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(
+                    'strategy',
+                    'Forbid multi-line whitespace or move the semicolon to the new line for chained calls.'
+            ))
+                ->setAllowedValues([self::STRATEGY_NO_MULTI_LINE, self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS])
+                ->setDefault(self::STRATEGY_NO_MULTI_LINE)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS === $this->configuration['strategy']) {
+            return $this->applyChainedCallsFix($tokens);
+        }
+
+        if (self::STRATEGY_NO_MULTI_LINE === $this->configuration['strategy']) {
+            return $this->applyNoMultiLineFix($tokens);
+        }
+    }
+
+    private function applyNoMultiLineFix(Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        foreach ($tokens as $index => $token) {
+            if (!$token->equals(';')) {
+                continue;
+            }
+
+            $previousIndex = $index - 1;
+            $previous = $tokens[$previousIndex];
+            if (!$previous->isWhitespace() || false === strpos($previous->getContent(), "\n")) {
+                continue;
+            }
+
+            $content = $previous->getContent();
+            if (("\n" === $content[0] || "\r" === $content[0]) && $tokens[$index - 2]->isComment()) {
```

Comment:
What exactly does this condition?

Maybe it should be compared to `$lineEnding` instead of covering both possible cases:

```php
$lineEnding === $content[0]
```

Reply:
I didn't want to touch the existing fixers logic, but maybe @SpacePossum can answer this? 

Changing it, breaks this test case https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.9/tests/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixerTest.php#L144

Reply:
> Changing it, breaks this test case...

It looks like that is 2nd argument of `WhitespacesFixerConfig` used there:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/df731ea139e00637056a806c55ff703c64ab0fec/tests/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixerTest.php#L135

Maybe the checks for the first sign only, so correct might be:

```php
// pseudo code, `strpos` function would be needed
Strings::startsWith($content,  $this->whitespacesFixerConfig->getLineEnding())
```

Reply:
Thanks for the example! I've updated it accordingly.